src/main/java/network/brightspots/rcv/Main.java

```diff
@@ -42,41 +49,53 @@ public static void main(String[] args) {
     Logger.setup();
     logSystemInfo();
 
-    // Determine if user intends to use the command-line interface, and gather args if so
-    boolean useCli = false;
-    List<String> argsCli = new ArrayList<>();
-    for (String arg : args) {
-      if (!useCli && arg.equals("-cli")) {
-        useCli = true;
-      } else if (useCli) {
-        argsCli.add(arg);
-      }
-    }
-
-    if (!useCli) {
-      // Launch the GUI
+    // Check if args contains string "--cli"
+    if (Arrays.stream(args).filter(arg -> arg.equals("--cli")).findAny().isEmpty()) {
+      // --cli not found. Launch the GUI
       launch(args);
     } else {
-      Logger.info("Tabulator is being used via the CLI.");
-      // Check for unexpected input
-      if (argsCli.size() == 0) {
+      Logger.info("Tabulator is being used via the CLI");
+
+      CommandLine cmd = parseArgsForCli(args);
+      String path = cmd.getOptionValue("cli");
+      String name = cmd.getOptionValue("name");
+      boolean convertToCdf = cmd.hasOption("convert-to-cdf");
+
+      boolean validNameProvided = false;
```

Comment:
I went ahead and redesigned this CLI logic so it handles corner cases more robustly. However, in doing so, I also found a critical flaw in this PR: the name that's entered and logged doesn't actually make it into the audit logs!

I think we need to make all this happen deeper in the app, i.e. immediately after this line:

```
Logger.info("Computer user name: %s", Utils.getUserName());
```

Also think it's probably fine if we don't prompt the user for their name when converting to CDF, but you could confirm with @chughes297 about this.

Reply:
Okay, I've made these changes: Convert to CDF doesn't require a name, and entering a blank name prevents tabulation.